File: src/main/java/no/nav/pensjon/innsyn/domain/Inntekt.java
package no.nav.pensjon.innsyn.domain;

public class Inntekt {

    private final String type;
    private final String status;
    private final int inntektsar;
    private final double belop;
    private final String rapportdato;
    private final String kilde;

    public Inntekt(String type, String status, int inntektsar, double belop, String rapportdato, String kilde) {
        this.type = type;
        this.status = status;
        this.inntektsar = inntektsar;
        this.belop = belop;
        this.rapportdato = rapportdato;
        this.kilde = kilde;
    }

    public String getType() {
        return type;
    }

    public String getStatus() {
        return status;
    }

    public int getInntektsar() {
        return inntektsar;
    }

    public double getBelop() {
        return belop;
    }

    public String getRapportdato() {
        return rapportdato;
    }

    public String getKilde() {
        return kilde;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/domain/popp/Inntekt.kt
package no.nav.pensjon.innsyn.domain.popp

import no.nav.pensjon.innsyn.domain.Domain
import no.nav.pensjon.innsyn.domain.popp.support.InntektStatus
import no.nav.pensjon.innsyn.domain.popp.support.InntektType
import no.nav.pensjon.innsyn.domain.popp.support.KildeType
import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "T_INNTEKT")
data class Inntekt(
        @Id
        @Column(name = "PERSON_ID")
        var personId: Int,
        @ManyToOne
        @JoinColumn(name = "K_INNTEKT_T")
        val inntektType: InntektType,
        @ManyToOne
        @JoinColumn(name = "K_INNTEKT_STATUS")
        val inntektStatus: InntektStatus,
        @Column(name = "INNTEKT_AR")
        val inntektsar: Int,
        @Column(name = "BELOP")
        val belop: Double,
        @Column(name = "PI_RAPPDATO")
        val rapportdato: LocalDate,
        @ManyToOne
        @JoinColumn(name = "K_Kilde_T")
        val kildeType: KildeType
) : Domain {
    @get:Transient
    val type
        get() = inntektType.dekode

    @get:Transient
    val status
        get() = inntektStatus.dekode

    @get:Transient
    val kilde
        get() = kildeType.dekode

    @get:Transient
    override val fields
        get() = setOf(::type, ::status, ::inntektsar, ::belop, ::rapportdato, ::kilde)
}